cumulusci/tasks/github/merge.py

```diff
@@ -15,57 +15,187 @@ class MergeBranch(BaseGithubTask):
+        },
     }
 
-    def _run_task(self):
-        repo = self.get_repo()
+    def _init_options(self, kwargs):
+        super(MergeBranch, self)._init_options(kwargs)
 
-        commit = self.options.get('commit', self.project_config.repo_commit)
-        branch_prefix = self.options.get('branch_prefix', self.project_config.project__git__prefix_feature)
-        source_branch = self.options.get('source_branch', self.project_config.project__git__default_branch)
+        if 'commit' not in self.options:
+            self.options['commit'] = self.project_config.repo_commit
+        if 'branch_prefix' not in self.options:
+            self.options['branch_prefix'] = self.project_config.project__git__prefix_feature
+        if 'source_branch' not in self.options:
+            self.options['source_branch'] = self.project_config.project__git__default_branch
+        if 'children_only' not in self.options:
+            self.options['children_only'] = False
+        elif self.options['children_only'] in [True, 'True']:
+            self.options['children_only'] = True
+
+    def _run_task(self):
+        self.repo = self.get_repo()
 
-        head_branch = repo.branch(source_branch)
+        head_branch = self.repo.branch(self.options['source_branch'])
         if not head_branch:
-            message = 'Branch {} not found'.format(source_branch)
+            message = 'Branch {} not found'.format(self.options['source_branch'])
             self.logger.error(message)
             raise GithubApiNotFoundError(message)
 
         # Get existing pull requests targeting a target branch
-        existing_prs = []
-        for pr in repo.iter_pulls(state='open'):
-            if pr.base.ref.startswith(branch_prefix):
-                existing_prs.append(pr.base.ref)
-        
-        targets = []
-        for branch in repo.iter_branches():
-            if not branch.name.startswith(branch_prefix):
-                self.logger.info('Skipping branch {}: does not match prefix {}'.format(branch.name, branch_prefix))
+        self.existing_prs = []
+        for pr in self.repo.iter_pulls(state='open'):
+            if pr.base.ref.startswith(self.options['branch_prefix']):
+                self.existing_prs.append(pr.base.ref)
+       
+        # Create list and dict of all target branches 
+        branches = []
+        branches_dict = {}
+        for branch in self.repo.iter_branches():
+            if branch.name == self.options['source_branch']:
+                if not self.options['children_only']:
+                    self.logger.debug('Skipping branch {}: is source branch'.format(branch.name))
+                    branches_dict[branch.name] = branch
+                    continue
+            if not branch.name.startswith(self.options['branch_prefix']):
+                if not self.options['children_only']:
```

Comment:
Is this `if` needed? Seems that children_only or not, we should be printing a message about skipping a branch due to not matching prefix.

Reply:
I skipped output here because in the case of a children_only merge you're merging from a parent feature branch to child feature branches.  It's impossible for a child branch to not have the feature prefix.  In the case of children_only merges, I wanted to just have it show the info on child branches it found rather than all the other branches.